source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -11,17 +11,20 @@ RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     : allowed_if_matched_(rules.action() ==
                           envoy::config::rbac::v2alpha::RBAC_Action::RBAC_Action_ALLOW) {
   for (const auto& policy : rules.policies()) {
-    policies_.emplace_back(policy.second);
+    policies_.insert(std::pair<std::string, PolicyMatcher>(policy.first, policy.second));
```

Comment:
Do we want to validate if there is a dup of policy ID?

Reply:
should be fine since rules.policies is map which guarantees no dup.